# oteodoro changes:
# Patch status:  testing
# Changed from Partial RELRO to Full RELRO
# For use in performance profiles

License header of ELF/Driver.cpp:

//===- Driver.cpp ---------------------------------------------------------===//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//

--- lld/ELF/Driver.cpp.orig	2022-01-20 13:31:59.000000000 -0800
+++ lld/ELF/Driver.cpp	2023-02-09 00:32:10.154967502 -0800
@@ -1184,10 +1184,10 @@ static void readConfigs(opt::InputArgLis
   config->zNodefaultlib = hasZOption(args, "nodefaultlib");
   config->zNodelete = hasZOption(args, "nodelete");
   config->zNodlopen = hasZOption(args, "nodlopen");
-  config->zNow = getZFlag(args, "now", "lazy", false);
+  config->zNow = getZFlag(args, "now", "lazy", false);         // Changed by oteodoro to disable Partial RELRO by default
   config->zOrigin = hasZOption(args, "origin");
   config->zPacPlt = hasZOption(args, "pac-plt");
-  config->zRelro = getZFlag(args, "relro", "norelro", true);
+  config->zRelro = getZFlag(args, "relro", "norelro", false);  // Changed by oteodoro to disable Partial RELRO by default
   config->zRetpolineplt = hasZOption(args, "retpolineplt");
   config->zRodynamic = hasZOption(args, "rodynamic");
   config->zSeparate = getZSeparate(args);
